alica_engine/include/engine/scheduler/Scheduler.h

```diff
@@ -53,7 +53,10 @@ class Scheduler
     JobId schedule(JobCb jobCb, std::optional<AlicaTime> repeatInterval = std::nullopt)
     {
         JobId jobId = _nextJobId++;
-        _jobQueue.emplace(jobId, std::move(jobCb), std::move(repeatInterval));
+        {
+            std::unique_lock<std::mutex> lock(_mutex);
+            _jobQueue.emplace(jobId, std::move(jobCb), std::move(repeatInterval));
```

Comment:
don't see why is this necessary? as a general thumb rule we should avoid acquiring mutex inside mutex.

Reply:
so the sequence that would fail I think is:
- Scheduler thread sees that's the `_jobQueue` is empty inside the `wait()` predicate i.e. `pop()` returns an empty optional
- Another thread adds a job to `_jobQueue` & calls `notify_one()`
- Scheduler thread misses the notify because it is still executing the `wait()` predicate & ends up waiting on the condition variable

Regarding the double lock, I am currently checking if we can get rid of the internal mutex in the queue.

Reply:
I got rid of the job queue's mutex